src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Triangle sides should be more than 0');
+    }
+
+    const maxSide: number = Math.max(a, b, c);
+    const perimeter: number = a + b + c;
+
+    if (maxSide >= perimeter - maxSide) {
+      throw new Error('Wrong sides for a triangle');
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
 
-export class Circle {
+    const area = Math.sqrt(
+      semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c),
+    );
 
+    return Math.round(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Triangle radius should be more than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Triangle radius should be more than 0');
+    }
+  }
+
+  getArea(): number {
+    return this.width * this.height;
```

Comment:
And what about rounding this value? Even if tests are passed, this method still works incorrect 🙂